From 1b5ff17bdd47bec61ec48b191e28f6bc1819557a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Thu, 2 May 2024 13:59:39 +0200
Subject: [PATCH 0861/1695] dts: stm32: add IWDG2 node in stm32mp21x soc
 device-tree
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add support for IWDG2 in stm32mp21x soc device-trees.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e9a394c6cab44a1ad39f8b63954e3c523ecd3422 dts: stm32: add IWDG2 node in stm32mp21x soc device-tree]
Change-Id: Ie5e58aefa0088381d3886d40946286af2af2bcfd
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/377760
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 5d5ce4be6..1be2f4a42 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -335,6 +335,15 @@
 				status = "disabled";
 			};
 
+			iwdg2: watchdog@44020000 {
+				compatible = "st,stm32mp1-iwdg";
+				reg = <0x44020000 0x400>;
+				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_IWDG2>, <&rcc LSI_CK>;
+				clock-names = "pclk", "lsi";
+				status = "disabled";
+			};
+
 			i2c3: i2c@46040000 {
 				compatible = "st,stm32mp25-i2c";
 				reg = <0x46040000 0x400>;
-- 
2.47.2

